### PR TITLE
Quality field

### PR DESCRIPTION
**Pull request by asaunier**
Related to https://github.com/c2corg/v6_api/issues/220 and https://github.com/c2corg/v6_common/pull/19

Currently the tests fail because
> sqlalchemy.engine.base.Engine: INFO: UPDATE guidebook.documents SET version=%(version)s, quality=%(quality)s WHERE guidebook.documents.document_id = %(guidebook_documents_document_id)s AND guidebook.documents.version = %(guidebook_documents_version)s
sqlalchemy.engine.base.Engine: INFO: {'quality': None, 'version': 2, 'guidebook_documents_version': 1, 'guidebook_documents_document_id': 1531}
sqlalchemy.engine.base.Engine: INFO: ROLLBACK
txn.139844722906880: DEBUG: abort
c2corg_api: DEBUG: debug_authorization of url http://localhost/waypoints/1531 (view name '' against context IntegrityError('(psycopg2.IntegrityError) null value in column "quality" violates not-null constraint\nDETAIL: Failing row contains (2, null, null, w, 1531, null).\n',)): Allowed (NO_PERMISSION_REQUIRED)

see https://github.com/c2corg/v6_common/pull/19#issuecomment-207486296

### CODE REVIEW

**Comment by tsauerwein** [comment]
> Do you mean I should make sure the quality attribute is actually set in https://github.com/c2corg/v6_api/blob/master/c2corg_api/models/document.py#L111 ?

When the field `quality` is not sent in an update request, it is like sending `"quality": null` which means that you explicitly want to set the value to null (which conflicts with the not-null constraint). If you don't want to change the value, you have to send the current value, like for all other fields.

I fixed the tests for you.

File: c2corg_api/models/document.py [hunk 1]
```diff
@@ -53,7 +54,8 @@ def redirects_to(self):
             Integer, ForeignKey(schema + '.documents.document_id'),
             nullable=True)
 
-    quality = Column(enums.quality_type)
+    quality = Column(
+        enums.quality_type, nullable=False, server_default=quality_types[1])
```
Comment (tsauerwein) [inline c2corg_api/models/document.py]: The default value should be 'stub'?

**Comment by tsauerwein** [comment]
Feel free to merge when you agree with my changes.

File: c2corg_api/models/document.py (hunk 1 above)
Comment (asaunier) [inline c2corg_api/models/document.py]: Yes, the client told me that the default value for migrated documents should be "medium" and "stub" for new documents (unless the contributor changes the quality value when saving the document - which is the encouraged behaviour).

**Comment by asaunier** [comment]
> I fixed the tests for you.

Wow, cool, thanks!

Then the quality field must be provided when saving the document, which does make sense since the field is available in the editing form (it is not possible to choose an empty value). And if it is missing for some reason, the API returns an error (because of the ``nullable=False`` constraint).

**Pull request merged**